pkg/skaffold/schema/v2beta29/upgrade.go

```diff
@@ -118,6 +118,17 @@ func (opu *OnePipelineUpgrader) upgradeOnePipeline(oldPipeline, newPipeline inte
 		if len(newPL.Render.Kustomize.Paths) == 0 {
 			newPL.Render.Kustomize.Paths = append(newPL.Render.Kustomize.Paths, ".")
 		}
+
+		// Inject Kubectl deployer to deploy kustomize manifests
+		kubectlDeployerWasInjected := false
+		if newPL.Deploy.KubectlDeploy == nil {
+			newPL.Deploy.KubectlDeploy = &next.KubectlDeploy{}
+			kubectlDeployerWasInjected = true
+		}
+
+		if err := mergeKustomizeIntoKubectlDeployer(newPL.Deploy.KubectlDeploy, oldPL.Deploy.KustomizeDeploy, kubectlDeployerWasInjected); err != nil {
```

Comment:
will doing `mergeKustomizeIntoKubectlDeployer(oldPL *Pipeline, newPL *next.Pipeline)` simplify this?

Reply:
Done! I moved all the logic to the new function I created, removing the third parameter used before. Let me know what do you think.